torch/csrc/distributed/c10d/init.cpp

```diff
@@ -957,6 +960,24 @@ Example::
     >>> store.set("first_key", "first_value")
     >>> # Should return "first_value"
     >>> store.get("first_key")
+)")
+          .def(
+              "_set_no_prefix",
```

Comment:
it might be worth naming these 'set/check_global' instead of '_no_prefix' to be clearer, although, arguably '_no_prefix' is actually clearer.  

(related to my comment above- should we avoid prefixes entirely?  should we add a prefix '/global' to the global keys to make that explicit in the key/value logging?) guess it doesn't make much difference.

Reply:
I tend to not to add a global prefix the reason is that this type of implicit prefix insertion is bug-prone because it requires users to have some tribal knowledge to use the store. Actually I think instead of layers of wrapping, we probably want to refactor PrefixStore to return what prefix it is using and return to users to loop-up. (And also flatten the PrefixWrap). Basically we want to give the option of whether to use and what prefix we are using to users and provide some transparency to users. WDYT?